src/cogent3/core/sequence.py

```diff
@@ -1941,6 +1942,7 @@ def __init__(
         self.step = step
         self._offset = offset
         self._seqid = seqid
+        self._seq_len = seq_len or len(self.seq)
```

Comment:
change this code to
```python
if seq_len is not None and seq_len != len(seq):
    raise AssertionError(f"{seq_len} != {len(self.seq)}")
```

Reply:
Should
 ```python
self._seq_len = seq_len or len(self.seq)
```
still be included? Or will `seq_len` always need to be provided?

Reply:
Good question, that should still be included!

The check is just to make sure if they are both provided that they're at least consistent.